### PR TITLE
add pug engine support

### PR DESCRIPTION
**Pull request by scottty881**

### CODE REVIEW

File: lib/mincer/engines/pug_engine.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ **/
+PugEngine.configure = function (opts) {
+  options = _.clone(opts);
+};
+
+
+// Render data
+PugEngine.prototype.evaluate = function (context, locals) {
+  if (this.nextProcessor && this.nextProcessor.name === 'JstEngine') {
+    // Use `compileClient` for Pug v1.0.0+, and `compile` for older versions
```
Comment (lastquestion) [inline lib/mincer/engines/pug_engine.js]: this comment seems spurious

File: lib/mincer.js [hunk 2]
```diff
@@ -288,6 +293,7 @@ Mincer.registerEngine('.jst',       Mincer.JstEngine);
 Mincer.registerEngine('.ejs',       Mincer.EjsEngine);
 Mincer.registerEngine('.hbs',       Mincer.HandlebarsEngine);
 Mincer.registerEngine('.jade',      Mincer.JadeEngine);
+Mincer.registerEngine('.pug',      Mincer.PugEngine);
```
Comment (puzrin) [inline lib/mincer.js]: align please

File: test/engines_test.js [hunk 3]
```diff
@@ -49,7 +49,6 @@ describe('Engines', function () {
         assert(asset.toString().match(/__output\.push/));
       });
     });
-
```
Comment (puzrin) [inline test/engines_test.js]: add witeline back

File: test/engines_test.js [hunk 4]
```diff
@@ -59,6 +58,16 @@ describe('Engines', function () {
         assert(asset.toString().match(/included/));
       });
     });
+    describe('with PUG backend', function () {
```
Comment (puzrin) [inline test/engines_test.js]: add witeline above please

File: lib/mincer.js [hunk 5]
```diff
@@ -70,6 +70,11 @@ Mincer.HandlebarsEngine   = require('./mincer/engines/handlebars_engine');
  **/
 Mincer.JadeEngine         = require('./mincer/engines/jade_engine');
 
+/**
+ *  Mincer.PugEngine -> PugEngine
+ *  Mincer.PugEngine -> PugEngine
```
Comment (ixti) [inline lib/mincer.js]: De-dup, please

File: test/engines_test.js [hunk 6]
```diff
@@ -49,7 +49,7 @@ describe('Engines', function () {
         assert(asset.toString().match(/__output\.push/));
       });
     });
-
```
Comment (ixti) [inline test/engines_test.js]: this should be empty line, not a line with couple of spaces. It just helps to visually separate describe blocks.

**Comment by ixti** [comment]
@scottty881 Thanks!

@puzrin once all changes addressed - I'm fine with this to be merged!

**Comment by scottty881** [comment]
@puzrin @ixti  not sure why my local linter didnt catch the spacing issue, but the build definitely did.
Let me know if theres anything else you need from me, would love to get this merged asap!
thanks!

**Comment by ixti** [comment]
Thanks!

**Pull request merged**